README.md

```diff
@@ -1,78 +1,77 @@
+[![build-test](https://github.com/jupyterhub/action-major-minor-tag-calculator/workflows/build-test/badge.svg)](https://github.com/jupyterhub/action-major-minor-tag-calculator/actions)
 
-Calculate major and minor semver tags, e.g. for tagging containers.
+This GitHub action can intelligently calculate and output a JSON formatted list of versions associated tags when a SemVer2 compliant tag is pushed go a GitHub repository.
 
-This will calculate major and minor prefix tags for when you want to provide users with a way to obtain the latest `major` or `major.minor` version.
+This action can be useful if a git tag like `1.2.3` is pushed to a GitHub repo and you want a GitHub Workflow's job to not only take action with regards to the pushed tag `1.2.3` specifically, but also to related tags like `1.2`, `1`, and `latest` assuming the pushed tag's version is new enough to represent those more general tags.
 
-Suffixes of the form `-[0-9]+` are supported since this is often used as a build number.
-Any other suffixes are treated as a pre-release, and are not supported.
+## Example output
 
-If this called on a branch instead of a tag just the branch-name is returned as the tag.
+The GitHub action's only output is named `tags` and is a JSON formatted list. See the example workflow below for details on how to convert the JSON formatted list to something else.
 
-## Examples
-
-Current and latest tag: `1.2.3`, returned tags: `[1.2.3, 1.2, 1, latest]`.
-
-Current tag `1.2.3` but the repository already contains a more recent tag `2.0.0`: return `[1.2.3, 1.2, 1]`.
-
-Branch `main`, return `[main]`.
+| Pushed reference | GitHub repo tags | `tags` output                        | Comment                                                                                |
+| ---------------- | ---------------- | ------------------------------------ | -------------------------------------------------------------------------------------- |
+| ``               | ...              | `"[]"`                               |
+| `main`           | ...              | `"[main]"`                           | Branches                                                                               |
+| `1.2.3`          | `1.2.0`          | `"[1.2.3, 1.2, 1, latest]"`          |
+| `1.2.3`          | `1.2.0`, `2.0.0` | `"[1.2.3, 1.2, 1]"`                  |
+| `1.2.3`          | `1.2.0`, `1.3.0` | `"[1.2.3, 1.2]"`                     |
+| `1.2.3`          | `1.2.0`, `1.2.5` | `"[1.2.3]"`                          |
+| `1.2.3-alpha.1`  | `1.2.0`          | `"[1.2.3-alpha.1]"`                  | Non-supported pre-release tags                                                         |
+| `1.2.3-4`        | `1.2.0`, `1.2.3` | `"[1.2.3-4, 1.2.3, 1.2, 1, latest]"` | Supported pre-release tags (build numbers, with a version suffix pattern of `-[0-9]+`) |
```

Comment:
Do you think it's worth dropping the mention of pre-release here to make it easier for people to understand? Technically according to semver `-N` is a pre-release, and that's therefore what the source uses, but we could treat that as an implementation detail?

Reply:
I agree these comments became a bit technical. I made a new attempt